File: challenges/5-cobaias/typescript/marcosvag/cobaias.ts
const promptSync = require('prompt-sync')();

const numeroDeCasos = parseInt(promptSync('Qual o numero de casos de testes? '));
const casos: { [caso: string]: number; } = {
    'R': 0,
    'S': 0,
    'C': 0,
    'TOTAL': 0,
};

function alocarCasos(numDeCasos: number): void {
    for (let e = 1; e <= numDeCasos; e++) {
        let caso = promptSync(`Caso ${e}: `);
        let numeroCobaiasCaso = parseInt(caso);
        casos[caso[caso.length - 1]] += numeroCobaiasCaso;
        casos['TOTAL'] += numeroCobaiasCaso;
    }
    analizarCasos();
}

function analizarCasos(): void {
    const casosTotal = casos['TOTAL'];
    const coelhosTotal = casos['C'];
    const ratosTotal = casos['R'];
    const saposTotal = casos['S'];

    const calcPorcentagem = (num: number, total: number = casosTotal): string => (num * 100 / total).toFixed(2);
    const coelhosPorcentagem = calcPorcentagem(coelhosTotal);
    const ratosPorcentagem = calcPorcentagem(ratosTotal);
    const saposPorcentagem = calcPorcentagem(saposTotal);

    console.log(`\nTotal: ${casosTotal} cobaias`);
    console.log(`Total de coelhos: ${coelhosTotal}`);
    console.log(`Total de ratos: ${ratosTotal}`);
    console.log(`Total de sapos: ${saposTotal}`);
    console.log(`Percentual de coelhos: ${coelhosPorcentagem} %`);
    console.log(`Percentual de ratos: ${ratosPorcentagem} %`);
    console.log(`Percentual de sapos: ${saposPorcentagem} %\n`);
}

alocarCasos(numeroDeCasos)

File: challenges/5-cobaias/java/Neg0023/Cobaias.java
import java.util.Scanner;

public class Cobaias {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(args[0]);
        int c = 0;
        int r = 0;
        int s = 0;
        int total=0;
        for (int i = 1; i <= n; i++) {

            int qtd = sc.nextInt();
            total+=qtd;
            char cobaia = sc.next().charAt(0);
            switch (cobaia) {
                case 'C' -> c += qtd;
                case 'R' -> r += qtd;
                case 'S' -> s += qtd;
            }
        }
        System.out.printf("Total: %d cobaias%n", total);
        System.out.printf("Total de coelhos: %d%n", c);
        System.out.printf("Total de ratos: %d%n", r);
        System.out.printf("Total de sapos: %d%n", s);
        double percentual = (double) c / total * 100;
        System.out.println("Percentual de coelhos: " + String.format("%.2f",percentual) + " %");
        percentual = (double) r / total * 100;
        System.out.println("Percentual de ratos: " + String.format("%.2f",percentual) + " %");
        percentual = (double) s / total * 100;
        System.out.println("Percentual de sapos: " + String.format("%.2f",percentual) + " %");
    }
}
